roles/edpm_download_cache/defaults/main.yml

```diff
@@ -26,3 +26,5 @@ edpm_download_cache_packages: true
 edpm_download_cache_container_images: true
 
 edpm_download_cache_running_services: "{{ edpm_services_override | default(edpm_services) }}"
+
+edpm_podman_auth_file: "~/.config/containers/auth.json"
```

Comment:
I don't think this is needed here, as it's already defined in edpm_podman. The var just needs to be set on tasks where it's used. Should we add it under a `vars` key in the download_cache playbook?

Reply:
We could do that. But I would much rather have the variable specified here.

While `edpm_podman` does use variable with the same name, and value, it uses it in different context. Meaning, it's used to create authfile, and store encoded credentials in it, with `podman login`. Obviously to log in to the registry as well, but that is given.

Whereas in `edpm_download_cache` it is used as source of credentials for authentication.[1]

Furthermore, by setting it in a playbook, we would loose benefits of argument specs[2], including argument validation and documentation. 

[1] https://docs.podman.io/en/latest/markdown/podman-pull.1.html#authfile-path
[2] https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec

Reply:
Are you saying this is meant to allow someone to create the auth file and use "source of credentials" from a different auth file?

You can't even do that since you named the vars the same.

I think you're making a distinction that doesn't have any practical application. The podman auth file is exactly that, it's not like someone would use one auth file to login, and then a different auth file to download images. Maybe I would buy it if you used a different var name here (edpm_download_cache_podman_auth_file).

The variable doesn't even follow our naming conventions here, that's why it starts with "edpm_podman", as it belongs in that role.

As for adding it in all the other download_cache.yml task files, I think I showed in an earlier review how that isn't needed. The default path will be checked. If the user really wants to override it, they can use the already existing var "auth_file" directly from the collection. A better way to address that would be to add a wrapper tasks file in our own edpm_podman role that uses the collection. That way, everything is forced through edpm_podman where the var is defined.

Reply:
FTR, I won't block on this as we need this in 18.0.1. I think we'll need to clean it up later though.